ocp_resources/resource.py

```diff
@@ -971,27 +971,25 @@ def wait_for_status(
                 **PROTOCOL_ERROR_EXCEPTION_DICT,
                 **DEFAULT_CLUSTER_RETRY_EXCEPTIONS,
             },
-            func=self.api.get,
-            field_selector=f"metadata.name=={self.name}",
-            namespace=self.namespace,
+            func=lambda: self.exists,
```

Comment:
this would work and you can also use self.instance.status here and compare it in line 980.